### PR TITLE
Feature/filter threads by role

### PR DESCRIPTION
**Pull request by puckybreg**
If this PR fixes a bug, you _must_ add test cases representative of the bug.
#### What's this PR do?
- Restricts the create-thread and reply-to-thread endpoints to the user scopes specified in the `MESSAGING_SERVICE_THREAD_SCOPES` env variable
#### How should this be manually tested?

- Update your env and test.env from the example env
   - `MESSAGING_SERVICE_THREAD_SCOPES=["admin", "clinician"]`
- Given a clinician user and a non-clinician user
- Create a thread with the clinician user (this should succeed)
- Create a thread with the non-clinician user (this should fail)
- Reply to a thread with the clinician user (this should succeed)
- Reply to a thread with the non-clinician user (this should fail)

- Run the tests and make sure everything passes (make sure your env.test is updated before doing this)

#### Screenshots (if appropriate):

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -5,7 +5,8 @@ LOG_LEVEL=info
 ## Messaging Microservice ##
 MESSAGING_SERVICE_PORT=4001
 
-MESSAGING_SERVICE_AUTOMATED_TEST_JWT=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InVzZXIwIiwiZW1haWwiOiJ0ZXN0QHRlc3QuY29tIiwiYWRtaW4iOnRydWV9.X_SzIXZ-oqEL67eB-fwFqFSumuFQVAqhgsmak1JLIWo
+MESSAGING_SERVICE_AUTOMATED_TEST_JWT = eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6NDUsInVzZXJuYW1lIjoidXNlcjAiLCJlbWFpbCI6InRlc3RAdGVzdC5jb20iLCJhZG1pbiI6dHJ1ZSwic2NvcGVzIjpbImNsaW5pY2lhbiIsImFkbWluIl0sImlhdCI6MTU0MjIwODIwMiwiZXhwIjozNjAwMDAwMDAwMDE1NDIyMDAwMDB9.e0sfocRLlARNT95R3P4rfS440t_r9K-QHQxDOlqYkB4
+MESSAGING_SERVICE_THREAD_SCOPES=["admin", "clinician"]
```
Comment (puckybreg) [inline .env.example:9]: Should I include `programAdministrator` in the list of scopes that can create and reply to threads?

File: server/routes/threads.route.js [hunk 2]
```diff
@@ -1,20 +1,39 @@
 import express from 'express';
 import passport from 'passport';
+import config from '../../config/config';
+import APIError from '/Users/jonah/Amida/services/amida-messaging-microservice/server/helpers/APIError.js';
```
Comment (orndorffgrant) [inline server/routes/threads.route.js]: Looks like an auto import here gave a computer specific path

File: .env.example (hunk 1 above)
Comment (orndorffgrant) [reply .env.example:9]: for the `.env.example` no. This can be configured in our orange deployment if we want that.

File: README.md [hunk 3]
```diff
@@ -227,6 +227,10 @@ The port this server will run on.
 
 This is the `amida-auth-microservice` JWT that is used by this repo's automated test suite when it makes requests.
 
+##### `MESSAGING_SERVICE_THREAD_SCOPES` (Required by test scripts)
+
+This is the `amida-auth-microservice` JWT that is used by this repo's automated test suite when it makes requests.
```
Comment (orndorffgrant) [inline README.md]: Can you add a quick description for this?

File: package.json [hunk 4]
```diff
@@ -83,6 +83,7 @@
     "eslint-config-airbnb-base": "7.1.0",
     "eslint-plugin-import": "1.16.0",
     "eslint-watch": "2.1.14",
+    "express-jwt-permissions": "false1.2.3",
```
Comment (orndorffgrant) [inline package.json]: looks like we ended up not using this, can it be removed?

File: server/tests/threads.integration.spec.js [hunk 5]
```diff
@@ -68,14 +68,15 @@ describe('Threads API:', () => {
...
+        };
         before(() => request(app)
             .post(`${baseURL}/threads`)
             .set('Authorization', `Bearer ${auth}`)
-            .send({
-                participants: ['user0', 'user1'],
-                message: 'test message',
-                topic: 'test topic',
-            })
+            .send(createThreadBody)
```
Comment (orndorffgrant) [inline server/tests/threads.integration.spec.js:79]: Fitting in code cleanup even on a tight deadline 👍

**Review by orndorffgrant — APPROVED** [review]
Works!

**Pull request merged**